### ISSUE TITLE
Sometimes "save" doesn't work 

### ISSUE THREAD

**Issue by danfloyd111** [jgraph/drawio-desktop#1857]
### Preflight Checklist

* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio/blob/dev/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Describe the bug**
Auto-save and manual save through the menu are inconsistent, sometimes they work and sometimes no. The shortcut "cmd+s" always fails without error messages.

**To Reproduce**
1. Do some modifications to an existing schema
2. Click on 'save' or wait for / trigger the auto-save
3. Wait for the message "All changes saved" to appear in the menu bar
4. Quit the application
5. Open the application again
6. The modifications are gone like nothing happened

**Expected behavior**
The modifications should be saved, if not possible for some reason an error message should be shown.

**draw.io version (In the Help->About menu of the draw.io editor):**

- draw.io DESKTOP version 24.7.8

**Desktop (please complete the following information):**
 - OS: macOS Sonoma 14.6.1
 - Browser Safari
 - Browser Version 17.6

**Additional context**
I'm using exclusively the desktop client, I did not test this functionality on the browser. I switched all the available "themes" but the issue persists. I added the details about my main browser (but I've also other browser installed on my machine).

**Comment by alderg** [comment]
Where did you save the file?

**Comment by danfloyd111** [comment]
> Where did you save the file?

In a local folder under my home directory

**Comment by alderg** [comment]
Is that directory synched with a cloud storage such as Google Drive, OneDrive or Dropbox?

**Comment by danfloyd111** [comment]
> Is that directory synched with a cloud storage such as Google Drive, OneDrive or Dropbox?

No, it's completely local, no iCloud either. Other than draw.io files, it contains a '.git' file to sync it with a remote repository, but I'm not performing any git ops in between.

I created a new diagram outside that folder, still completely local and without binding this to any git repository: the issue is still here. The auto-save seems more consistent, it works 80% of the times. Saving with cmd+s is not working at all (no messages are shown) while saving through the menu triggers the "all changes are saved" messages but after quitting the application the modifications disappear.

**Comment by alderg** [comment]
Are you using the desktop version or the web app (in a browser)? Please include all the version information requested in the original issue template.

**Comment by danfloyd111** [comment]
> Are you using the desktop version or the web app (in a browser)? Please include all the version information requested in the original issue template.

Sorry, I thought I did that by leaving blank the "browser lines". Fixing it now.
I'm using the desktop version 24.7.8.

**Comment by danfloyd111** [comment]
Edit: I was confused reading "If you delete part/all or miss parts out your issue will be closed." in the instruction, this is the reason why I left some fields not filled. 
Following the example of other reports, I cleaned the report. Please let me know if now it is ok.

**Comment by alderg** [comment]
Yes, all good. But it will be difficult to provide any help since you seem to be the only person to have this issue. We'd need more information to help us narrow down the problem. For example, what is the size of the file you are trying to save?

**Comment by danfloyd111** [comment]
> Yes, all good. But it will be difficult to provide any help since you seem to be the only person to have this issue. We'd need more information to help us narrow down the problem. For example, what is the size of the file you are trying to save?

The file I'm working on is about 30MB, but I was able to reproduce the issue on a new schema created just for testing purposes which is about 4KB.

I can record a video of the issue if it can be helpful.
Thanks for your super quick replies!

More info: it doesn't seems to be a "I'm too quick to quit issue". This time I created a test file with a text field and triggered the auto-save feature, then I closed the schema. I opened the schema again and did a quick modification of the text field, then I saved the file through the menu [screenshot 1], then after the "all changes saved" badge appeared on the top I quit the application (cmd+q). After I re-opened the application, the modification I did moments before was gone [screenshot 2]. Hitting cmd+s doesn't even trigger the "all changes saved" badge.
 
Screenshot 1
<img width="1387" alt="Screenshot 2024-09-03 at 16 34 20" src="https://github.com/user-attachments/assets/6018809a-b0a6-4db7-8bd9-612186451031">

Screenshot 2
<img width="1381" alt="Screenshot 2024-09-03 at 16 34 33" src="https://github.com/user-attachments/assets/3939e161-f7c4-4cd4-b973-7a3adde5424e">

**Comment by skaufman** [comment]
Seeing same issue with MacOS desktop drawio v24.7.17: manual save and autosave both not working. This is a new issue from earlier versions.

**Comment by m-mohamedin** [comment]
Is this only when editing a label text?

**Comment by danfloyd111** [comment]
No, it is not related to the element type, at least for me.On 27 Dec 2024, at 17:07, Mohamed Mohamedin ***@***.***> wrote:﻿
Is this only when editing a label text?

**Comment by fgc0109** [comment]
My work over the past two days has all vanished. it tell me saved, but not success! maybe not finish 

file become a 512 kb rubbish，only give me a message "Error not a diagram file  (error on line 1 at column 524285: Premature end of data in tag diagram line 1)"

`
<mxfile host="Electron" agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) draw.io/25.0.2 Chrome/128.0.6613.186 Electron/32.2.5 Safari/537.36" compressed="true" version="25.0.2"><diagram name="第 1 页" id="tDCwUokWXdFA1eB2x_8P">jHzXztxckuTT/JcL0JtLeu89bxb03pN ........
`

I will not continue using this software anymore

**Comment by davidjgraph** [comment]
Still happening on 26.2.15?

**Comment by MichaelSjostrom** [comment]
> Still happening on 26.2.15?

Yes. It happened to med with many different files.

**Comment by C0KERNEL** [comment]
Experienced this in draw.io Desktop v28.1.2 on Windows 11 Pro version 24H2.

**Comment by OpenTY** [comment]
我也遇到了这个问题,无法保存文件.同样的软件在其他电脑上保存正常,但在Ultra5 225H windows上大概率保存失败,偶尔会成功

**Comment by OpenTY** [comment]
> 我也遇到了这个问题，无法保存文件。同样的软件在其他电脑上保存正常，但在Ultra5 225H windows上大概率保存失败，偶尔会成功

软件错误日志已上传
[2026-05-11 20:16:37.575] [error] Error: Error: net::ERR_CONNECTION_RESET
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:2:134205)
    at SimpleURLLoaderWrapper.emit (node:events:508:28)

[2026-05-11 20:16:37.630] [error] @error@
 Error: net::ERR_CONNECTION_RESET
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:2:134205)
    at SimpleURLLoaderWrapper.emit (node:events:508:28)
